File: binderBack/src/main/java/com/binder/entity/GiveAway.java
package com.binder.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "GiveAway")
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class GiveAway {
    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "giveaway_seq")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "book_id", nullable = false)
    private Book book;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "description")
    private String description;

    @Lob
    @Column(name = "photo")
    private byte[] photo;
}

File: Binder/app/src/main/java/com/example/binder/model/Giveaway.kt
package com.example.binder.model

import com.squareup.moshi.Json

data class Giveaway(
    @field:Json(name = "id") val id: Long,
    @field:Json(name = "userId") val userId: Long,
    @field:Json(name = "description") val description: String,
    @field:Json(name = "book") val book: Book,
    @field:Json(name = "photo") val photo: String?,
)